### PR TITLE
[WIP] Add set_memory_request_to_limit option

### PR DESCRIPTION
**Pull request by zerg-junior**
Add `set_memory_request_to_limit` option. Also adds a few clarifications addresses in #403

### CODE REVIEW

File: manifests/configmap.yaml [hunk 1]
```diff
@@ -15,6 +15,7 @@ data:
   secret_name_template: '{username}.{cluster}.credentials'
   super_username: postgres
   enable_teams_api: "false"
+  set_memory_request_to_limit: "true"
```
Comment (Jan-M) [inline manifests/configmap.yaml]: why change this to true when we default to false?

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -110,6 +110,29 @@ func (c *Controller) initOperatorConfig() {
 	c.opConfig = config.NewFromMap(configMapData)
 	c.warnOnDeprecatedOperatorParameters()
 
+	if c.opConfig.SetMemoryRequestToLimit {
+
+		isSmaller, err := util.RequestIsSmallerThanLimit(c.opConfig.DefaultMemoryRequest, c.opConfig.DefaultMemoryLimit)
```
Comment (zerg-junior) [inline pkg/controller/controller.go]: I am unhappy with the way this function handles errors.Does anyone have  a better idea on how to re-write it ?

File: docs/reference/operator_parameters.md [hunk 3]
```diff
@@ -221,6 +221,9 @@ CRD-based configuration.
   memory limits for the postgres containers, unless overridden by cluster-specific
   settings. The default is `1Gi`.
 
+* **set_memory_request_to_limit**
+  Set `memory_request` to `memory_limit` for all Postgres clusters (the default value is also increased). This prevents certain cases of memory overcommitment at the cost of overprovisioning memory and potential scheduling problems for Postgres pods with high memory limits due to the lack of memory on Kubernetes cluster nodes. This affects sidecar containers as well. The default is `false`.
```
Comment (Jan-M) [inline docs/reference/operator_parameters.md]: This affects all possible containers, not only Postgres, might need a word on that.

**Comment by zerg-junior** [comment]
:-1: 
`RequestIsSmallerThanLimit` incorrectly processes clusters manifests w/o explicit memory requirements/limits

**Comment by jrake-revelant** [comment]
hey folks, is there an ETA on getting this to `stups-test` ?

**Pull request merged**